inference-engine/src/mkldnn_plugin/nodes/mkldnn_deconv_node.cpp

```diff
@@ -79,23 +85,119 @@ MKLDNNDeconvolutionNode::MKLDNNDeconvolutionNode(const std::shared_ptr<ngraph::N
             paddingL = groupConvBackprop->get_pads_begin();
             paddingR = groupConvBackprop->get_pads_end();
         }
+        for (int i = 0; i < dilation.size(); i++) {
+            kernel.push_back(weightDims[withGroups + 2 + i]);
+        }
     } else {
         IE_THROW(NotImplemented) << errorMessage;
     }
 }
 
+InferenceEngine::Blob::Ptr MKLDNNDeconvolutionNode::createWeiBlobAsIO(InferenceEngine::SizeVector dims) {
+    auto checkSize = [](size_t dst_size, size_t src_size) {
+        if (dst_size < src_size) {
+            IE_THROW() << "Cannot create internal buffer. Buffer can be overrun.";
+        }
+    };
+
+    auto constNode = std::dynamic_pointer_cast<MKLDNNInputNode>(getParentEdgeAt(1)->getParent());
+    if (!constNode)
+        IE_THROW() << "Cannot cast const input node for node " << getName() << ".";
+    auto blb = constNode->getConstBlob();
+    if (!blb)
+        IE_THROW() << "Cannot get const weights blob for node " << getName() << ".";
+
+    // WA: In int8 case, we are processing weights using internal blob.
+    // So we disconnect constant node containing weights from the graph and then don't use it.
+    if (getParentEdges().size() == 3) {
+        removeEdge(getParentEdgeAt(2));
+        inDims.erase(inDims.begin() + 2);
+    }
+    removeEdge(getParentEdgeAt(1));
+    inDims.erase(inDims.begin() + 1);
+
+    InferenceEngine::SizeVector dimsForBlockedDesc{dims};
+    std::swap(dimsForBlockedDesc[withGroups + 0], dimsForBlockedDesc[withGroups + 1]);
+
+    InferenceEngine::SizeVector orderForBlockedDesc;
+    if (withGroups) {
+        orderForBlockedDesc = {0, 2, 1};
+    } else {
+        orderForBlockedDesc = {1, 0};
+    }
+    for (int i = 2 + withGroups; i < dimsForBlockedDesc.size(); i++)
+        orderForBlockedDesc.push_back(i);
+
+    BlockingDesc blkDesc(dimsForBlockedDesc, orderForBlockedDesc);
+    InferenceEngine::TensorDesc tensorDesc(blb->getTensorDesc().getPrecision(), dims, blkDesc);
+
+    auto fillInternalBlob = [&](char *data, size_t intBuffSize) {
+        size_t offset = blb->byteSize();
+        checkSize(intBuffSize, offset);
```

Comment:
the `checkSize` lambda is used once. Suggest inline the code.

Reply:
fixed, thanks